chainerio/filesystems/hdfs.py

```diff
@@ -30,36 +34,81 @@ def _parse_klist_output(output):
         return None
 
 
+def _parse_principal_name_from_keytab(output):
+    output_array = output.split('\n')
+    if len(output_array) < 4:
+        return None
+
+    principle_str = output_array[3]
+    keytab_principle_pattern = re.compile(
+        r'\s+\d+ (?P<username>.+)@(?P<service>.+)')
+    ret = keytab_principle_pattern.match(principle_str)
+    if ret:
+        pattern_dict = ret.groupdict()
+        return pattern_dict['username']
+    else:
+        return None
+
+
+def _get_principal_name_from_keytab():
+    keytab_path = os.getenv("KRB5_KTNAME")
```

Comment:
I don't this line is needed, but just running `klist -k` is enough. Because not only it automatically refers to `KRB5_KTNAME`but follows default fallback to `/etc/krb5.keytab` .